pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Markers.java

```diff
@@ -309,24 +306,24 @@ private static ByteBuf newTxnMarker(MarkerType markerType, long sequenceId, long
         MessageMetadata msgMetadata = msgMetadataBuilder.build();
 
         ByteBuf payload;
-        if (messageIdData.isPresent()) {
-            PulsarMarkers.TxnCommitMarker commitMarker = PulsarMarkers.TxnCommitMarker.newBuilder()
-                                                                                      .setMessageId(messageIdData.get())
-                                                                                      .build();
-            int size = commitMarker.getSerializedSize();
-            payload = PooledByteBufAllocator.DEFAULT.buffer(size);
-            ByteBufCodedOutputStream outStream = ByteBufCodedOutputStream.get(payload);
-            commitMarker.writeTo(outStream);
-        } else {
-            payload = PooledByteBufAllocator.DEFAULT.buffer();
-        }
+        PulsarMarkers.TxnCommitMarker.Builder commitMarkerBuilder = PulsarMarkers.TxnCommitMarker.newBuilder();
+
+        messageIdDataList.ifPresent(commitMarkerBuilder::addAllMessageId);
+        PulsarMarkers.TxnCommitMarker commitMarker = commitMarkerBuilder.build();
+        int size = commitMarker.getSerializedSize();
+        payload = PooledByteBufAllocator.DEFAULT.buffer(size);
+        ByteBufCodedOutputStream outStream = ByteBufCodedOutputStream.get(payload);
+        commitMarker.writeTo(outStream);
 
         try {
             return Commands.serializeMetadataAndPayload(ChecksumType.Crc32c, msgMetadata, payload);
         } finally {
             payload.release();
             msgMetadata.recycle();
             msgMetadataBuilder.recycle();
+            if (commitMarkerBuilder != null) {
```

Comment:
The commitMarkerBuilder always not null.

Reply:
I'll fix this.